Enabling HTTP Basic Auth

adamgoose opened the pull request:
- Enabling HTTP Basic Auth
- Tweaking argument validation
- Setting reasonable defaults
- Rebasing now

cedvan commented on README.md:
```diff
@@ -106,6 +106,9 @@ Below is the complete list of available options that can be used to customize yo
 - **TORAN_TRACK_DOWNLOADS**: Track private package installs, set to `true` to get an install log in `/data/toran-proxy/logs/downloads.private.log`. Defaults to `false`
 - **TORAN_MONO_REPO**: Set to `true` for switch to a monorepo instead of dual repo model. Defaults to `false`
 - **PHP_TIMEZONE**: Configure timezone PHP. Default `Europe/Paris`.
+- **TORAN_AUTH_ENABLE**: Set to `true` to enable HTTP Basic Authentication. When enabled, `TORAN_AUTH_USER` and `TORAN_AUTH_PASSWORD` are required.
```
miss : Default to `false`.

cedvan commented on scripts/config.sh:
```diff
@@ -23,6 +23,9 @@ TORAN_TOKEN_GITHUB=${TORAN_TOKEN_GITHUB:-false}
+TORAN_AUTH_ENABLE=${TORAN_AUTH_ENABLE:-false}
+TORAN_AUTH_USER=${TORAN_AUTH_USER:-false}
+TORAN_AUTH_PASSWORD=${TORAN_AUTH_PASSWORD:-false}
 
```
Please add verification format to new params

cedvan commented on scripts/config.sh:
```diff
@@ -23,6 +23,9 @@ TORAN_TOKEN_GITHUB=${TORAN_TOKEN_GITHUB:-false}
 TORAN_MONO_REPO=${TORAN_MONO_REPO:-false}
 TORAN_SECRET=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1)
+TORAN_AUTH_ENABLE=${TORAN_AUTH_ENABLE:-false}
+TORAN_AUTH_USER=${TORAN_AUTH_USER:-false}
```
default must be : `toran`

cedvan commented on scripts/config.sh:
```diff
@@ -23,6 +23,9 @@ TORAN_TOKEN_GITHUB=${TORAN_TOKEN_GITHUB:-false}
 TORAN_SECRET=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1)
+TORAN_AUTH_ENABLE=${TORAN_AUTH_ENABLE:-false}
+TORAN_AUTH_USER=${TORAN_AUTH_USER:-false}
+TORAN_AUTH_PASSWORD=${TORAN_AUTH_PASSWORD:-false}
```
default must be : `toran`

cedvan commented on scripts/install/nginx.sh:
```diff
@@ -1,5 +1,40 @@
+echo "Detecting HTTP Basic Authentication Configuration"
+if [ "${TORAN_AUTH_ENABLE}" != "false" ]; then
+
+    if [ "${TORAN_AUTH_USER}" == "false" ]; then
```
check useless with default `toran` and format verification in `config.sh`

cedvan commented on scripts/install/nginx.sh:
```diff
@@ -1,5 +1,40 @@
+        exit 1
+    fi
+
+    if [ "${TORAN_AUTH_PASSWORD}" == "false" ]; then
```
check useless with default `toran` and format verification in `config.sh`

cedvan commented on README.md:
```diff
@@ -106,6 +106,9 @@ Below is the complete list of available options that can be used to customize yo
 - **TORAN_MONO_REPO**: Set to `true` for switch to a monorepo instead of dual repo model. Defaults to `false`
 - **PHP_TIMEZONE**: Configure timezone PHP. Default `Europe/Paris`.
+- **TORAN_AUTH_ENABLE**: Set to `true` to enable HTTP Basic Authentication. When enabled, `TORAN_AUTH_USER` and `TORAN_AUTH_PASSWORD` are required.
+- **TORAN_AUTH_USER**: Configure the HTTP Basic Authentication Username
```
add : Default to `toran`

cedvan commented on README.md:
```diff
@@ -106,6 +106,9 @@ Below is the complete list of available options that can be used to customize yo
 - **PHP_TIMEZONE**: Configure timezone PHP. Default `Europe/Paris`.
+- **TORAN_AUTH_ENABLE**: Set to `true` to enable HTTP Basic Authentication. When enabled, `TORAN_AUTH_USER` and `TORAN_AUTH_PASSWORD` are required.
+- **TORAN_AUTH_USER**: Configure the HTTP Basic Authentication Username
+- **TORAN_AUTH_PASSWORD**: Configure the HTTP Basic Authentication Password
```
add : Default to `toran`

cedvan commented on scripts/config.sh:
```diff
@@ -90,3 +93,15 @@ if [[ ! "${TORAN_CRON_TIMER_DAILY_TIME}" =~ ^[0-9]{2}:[0-9]{2}$ ]]; then
     echo "  Variable TORAN_CRON_TIMER_DAILY_TIME isn't valid ! (Format accepted : HH:MM)"
     exit 1
 fi
+
```
missing : 

```
# Checking Toran auth enable
if [ "${TORAN_AUTH_ENABLE}" != "true" ] && [ "${TORAN_AUTH_ENABLE}" != "false" ]; then
    echo "ERROR: "
    echo "  Variable TORAN_AUTH_ENABLE isn't valid ! (Values accepted : true/false)"
    exit 1
fi
```

cedvan commented:
Can you rebase to one commit please ?

adamgoose committed:
Enabling HTTP Basic Auth

- Enabling HTTP Basic Auth
- Tweaking argument validation
- Setting reasonable defaults
- Rebasing now

cedvan commented:
And just add small description with list changes, and i'll merge

adamgoose commented:
I did put something in the changelog. Was there anything else/more specific you're looking for?

cedvan commented:
Same changelog is ok :wink:

cedvan commented:
No previous changelog was good, rollback changelog and just copy previous text from changelog in merge request description :smile:

cedvan commented:
Thank for contributing :+1:

adamgoose commented:
Thanks for merging! :) Is this going to build and tag as the latest 1.5.3 on Docker Hub?
